pygit2/repository.py

```diff
@@ -1159,14 +1160,23 @@ def get(self, key):
 
     def __iter__(self):
         for branch_name in self._repository.listall_branches(self._flag):
-            yield branch_name
+            if self._valid(self[branch_name]):
```

Comment:
I would rather call `self._valid(branch_name)` to avoid a lookup in the case that `self._commit is None`, which is probably the most common case.

Reply:
Good point, but I don't want to incur the lookup in the case of `__getitem__` where we already have a branch reference object.   Check my latest commit for what I think is a good compromise.